temporal-sdk/src/main/java/io/temporal/common/context/ContextPropagator.java

```diff
@@ -31,71 +31,75 @@
- *         for (Map.Entry<String, String> entry : contextMap.entrySet()) {
- *             MDC.put(entry.getKey(), entry.getValue());
- *         }
+ *   public void setCurrentContext(Object context) {
+ *     Map<String, String> contextMap = (Map<String, String>) context;
+ *     for (Map.Entry<String, String> entry : contextMap.entrySet()) {
+ *       MDC.put(entry.getKey(), entry.getValue());
  *     }
+ *   }
  *
- *     public Map<String, byte[]> serializeContext(Object context) {
- *         Map<String, String> contextMap = (Map<String, String>)context;
- *         Map<String, byte[]> serializedContext = new HashMap<>();
- *         for (Map.Entry<String, String> entry : contextMap.entrySet()) {
- *             serializedContext.put(entry.getKey(), entry.getValue().getBytes(StandardCharsets.UTF_8));
- *         }
- *         return serializedContext;
+ *   public Map<String, Payload> serializeContext(Object context) {
+ *     Map<String, String> contextMap = (Map<String, String>) context;
+ *     Map<String, Payload> serializedContext = new HashMap<>();
+ *     for (Map.Entry<String, String> entry : contextMap.entrySet()) {
+ *       serializedContext.put(entry.getKey(), DataConverter.getDefaultInstance().toPayload(entry.getValue()).get());
  *     }
+ *     return serializedContext;
+ *   }
  *
- *     public Object deserializeContext(Map<String, Payload> context) {
- *         Map<String, String> contextMap = new HashMap<>();
- *         for (Map.Entry<String, byte[]> entry : context.entrySet()) {
- *             contextMap.put(entry.getKey(), new String(entry.getValue(), StandardCharsets.UTF_8));
- *         }
- *         return contextMap;
+ *   public Object deserializeContext(Map<String, Payload> context) {
+ *     Map<String, String> contextMap = new HashMap<>();
+ *     for (Map.Entry<String, Payload> entry : context.entrySet()) {
+ *       contextMap.put(entry.getKey(), DataConverter.getDefaultInstance().fromPayload(entry.getValue(), String.class, String.class));
  *     }
+ *     return contextMap;
+ *   }
  * }
  * }</pre>
  *
- * To set up the context propagators, you must configure them when: <br>
- * Creating WorkflowClient:
+ * To set up the context propagators, you must configure them during creation of WorkflowClient on
+ * both client/stubs and worker side: <br>
  *
  * <pre>{@code
  * WorkflowClientOptions options = WorkflowClientOptions.newBuilder()
  *                 .setContextPropagators(Collections.singletonList(new MDCContextPropagator()))
  *                 .build();
  * WorkflowClient client = WorkflowClient.newInstance(service, factoryOptions);
+ *
+ * Workflow workflow = client.newWorkflowStub(Workflow.class,
+ *                WorkflowOptions.newBuilder().setTaskQueue("taskQueue").build());
+ * //or
+ * WorkerFactory workerFactory = WorkerFactory.newInstance(client);
  * }</pre>
  *
  * <br>
- * If you want to have override the {@code ContextPropagator} instances for your activities, you can
```

Comment:
this section is gone, because it actually pushes users to incorrect understanding and usage of ActivityOptions#contextPropagators